qiskit/assembler/assemble_schedules.py

```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
     Raises:
         QiskitError: If the instructions do not satisfy the measurement map.
     """
+    sorted_inst_map = sorted(instruction_map.items(), key=lambda item: item[0])
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
+    all_measured_qubits = [inst.channel.index for _, insts in sorted_inst_map
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
```

Comment:
`instr` isn't an instruction, it's a tuple with information about timing (`start_time, duration`)